packages/smithy-http/src/smithy_http/aio/interfaces/__init__.py

```diff
@@ -12,14 +12,23 @@
     HTTPRequestConfiguration,
 )
 
+Config = TypeVar("Config", contravariant=True)
+
+
+class EndpointParametersProtocol(Protocol[Config]):
+    @classmethod
+    def build(cls, config: Config) -> Self:
+        raise NotImplementedError()
```

Comment:
As of python 3.12 we have proper generic declarations and inferred variance, so `TypeVar` is not needed.

```suggestion
class EndpointParameters[C](Protocol):
    @classmethod
    def build(cls, config: C) -> Self:
        raise NotImplementedError()
```